src/eventio/base.py

```diff
@@ -125,7 +125,11 @@ def __next__(self):
 
     def peek(self):
         if self.next is None:
-            self.next = next(self)
+            try:
+                self.next = next(self)
+            except (StopIteration, EOFError, IOError):
+                self.next = None
```

Comment:
Note to self: this should probably store that an error has happened and then `__next__` should fail in that case.